### ISSUE TITLE
[FEATURE REQUEST] To have "TATE mode" (rotated screen) on MAME attached to GAME OVERRIDE and not to universal core settings

### ISSUE THREAD

**Issue by giovanifaganello** [libretro/RetroArch#11279]
## Description

There is a awesome and useful setting called TATE MODE on MAME cores. If you have a screen capable of rotation (vertical/horizontal) you can enjoy the shooting games on its full glory. IMPORTANT: using scaling as CUSTOM, INTEGER ON and sizes of 3X to have pixel perfect screen.

### Expected behavior

When you load a MAME title, no matter what MAME core you use, you can call the in-game menu, go to OPTIONS and toggle the TATE MODE, so it will rotate your screen vertically, but — first bug — it rotates the display itself but does not "flip" the WIDTH and HEIGHT sizes, so, you need to go all way back to VIDEO > SCALING and change the values there.

### Actual behavior

However, upon exiting the game (closing content) this setting will be permanent applied to the very core itself, so, applied to all its games.

I don't know the math for it, but if it could be applied just to the game we have open ITSELF it could be awesome. Also, FLIP the WIDTH x HEIGHT when TATE is enabled, and, of course, update the very own game override too.

### Steps to reproduce the bug

0. Set your video as INTEGER ON, CUSTOM and sizes 3X.
1. Load a mame game you want to play vertically
2. Go to options
3. Turn on the TATE option
4. Check the game, it will be rotated, but distorted (w x h have not being "flipped")
5. Exit the game, and open another one
6. This game will be also rotated... you need to go to option and turn off tate again.

### Version/Commit

Any version as 1.8.4, 1.8.5 and 1.9.0

### Environment information

- OS: PlayStation Classic
- Compiler: https://www.reddit.com/r/PlaystationClassic/comments/ikhv3v/retroboot_12_the_lightweight_alternative/

**Comment by bslenul** [comment]
Load your game, turn ON TATE mode, then on top of the Options menu you should have an option "Create game options file", which will save current core options for that game only.

**Comment by giovanifaganello** [comment]
Oh my. It works! Thanks bslenul!
I am pretty sure this was not working on the 1.8.5 version... or I was doing something wrong.

But, anyway — if you have video on CUSTOM+INTEGER ON+SIZE 3X — upon setting TATE mode, the values of WIDTH x HEIGHT should be flipped with the screen as well, otherwise the game screen gets distorted.

Don't you agree?

**Comment by giovanifaganello** [comment]
> Load your game, turn ON TATE mode, then on top of the Options menu you should have an option "Create game options file", which will save current core options for that game only.

Just checked and saw that now we have a GAME-NAME.OPT file created, containing core options for THAT game exclusively! This is something new to 1.9.0, right?

It is wonderful!

**Comment by bslenul** [comment]
> But, anyway — if you have video on CUSTOM+INTEGER ON+SIZE 3X — upon setting TATE mode, the values of WIDTH x HEIGHT should be flipped with the screen as well, otherwise the game screen gets distorted.

Hm I guess it would make sense yeah, however I'm not a dev so I could be wrong but I'd say that would be a feature request for the core, not for RetroArch. I mean I guess it would be the core's job to tell the frontend to swap width/height.

> Just checked and saw that now we have a GAME-NAME.OPT file created, containing core options for THAT game exclusively! This is something new to 1.9.0, right?

Haha nope, it's been there for at least 3 years :D I don't know exactly when it was implemented but it already existed in 1.6.0 :p

**Comment by giovanifaganello** [comment]
> Hm I guess it would make sense yeah, however I'm not a dev so I could be wrong but I'd say that would be a feature request for the core, not for RetroArch. I mean I guess it would be the core's job to tell the frontend to swap width/height.

Yes, it makes sense... A core's job. Well, is there a GiHub where I could post my suggestion directly to the "MAME's core people"? (I am using RA on PS Classic...

> Haha nope, it's been there for at least 3 years :D I don't know exactly when it was implemented but it already existed in 1.6.0 :p

I really don't know what I was doing then. Perhaps it could be related to the MAME cores on PS Classic. Glad it works now!

**Comment by bslenul** [comment]
Depends which MAME core you're talking about (probably 2003 or 2010 on the PSC?), there are a few: https://github.com/libretro?q=mame&type=&language= :p

**Comment by giovanifaganello** [comment]
> If it's KMFD MAME2003 "Xtreme" variant: https://github.com/KMFDManic/mame2003-xtreme/issues

Bingo! That's exactly the core!
Thanks!

**Comment by giovanifaganello** [comment]
People, after talking to Grant on KMFD Mame2003-xtreme core GitHub, he told me that it is RetroArch is the one responsible for the geometry of the screen, set up on a GAME.CFG file for MAME.

Do you think it is possible then, something like this scenario:

— If you have INTEGER SCALING  = TRUE
— If the game has its own GAME.CFG file
— If the GAME.CFG has WIDTH = X and HEIGHT = Y
— When TATE is TRUE, upon loading the game, SWAP screen coordinates WIDTH = Y  and HEIGHT = X?
— If TATE is set back to FALSE
— Loads the game normally, as found in its CFG, W x H

A bit more of information:

Yes, my big LG 42" rotates on the wall easily. Some people out there have setups like that, even smaller monitors (Dell) have the physical feature of rotating to vertical (very useful for people who work with publishing and graphic design).

For me, TATE is totally intended for people who really rotate their TVs or monitors, so this is why I am here now.

But, finally, this is just a humble request. I have no idea of the odds and ends to such a change. But it would be awesome if it is possible to be implemented!

As always, THANK YOU for everything on RetroArch.
I love it from the bottom of my heart!

**Comment by curif** [comment]
It's possible to solve the issue by adding 
`buf_status_cb.callback = &retro_audio_buff_status_cb; `
In the line 142 of `mame2003.c`
